src/Microsoft.Security.Utilities.Core/HighPerformanceScanner.cs

```diff
@@ -0,0 +1,505 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.Security.Utilities;
+
+#if NET8_0_OR_GREATER
+using System.Buffers;
+using StringInput = System.ReadOnlySpan<char>;
+#else
+using StringInput = string;
+#endif
+
+/// <summary>
+/// A detection from <see cref="HighPerformanceScanner"/>.
+/// </summary>
+internal readonly record struct HighPerformanceDetection
```

Comment:
>HighPerformanceDetection [](http://example.com/codeflow?start=32&length=24)

NIT: what's your care factor around the `single type, single file` standard? i usually conform to this.

Reply:
I don't like the standard, but my care factor is not high enough to push back.

Reply:
Again, my time away from .NET influenced me here.

Reply:
Done. Thank you, refactoring -> move to file. I will do it pre-checkin next time. :)